signer/src/transaction_signer.rs

```diff
@@ -557,28 +540,28 @@ where
                     return Ok(());
                 }
 
-                let id = StateMachineId::from(bitcoin_chain_tip);
-                self.relay_message(id, msg.txid, &msg.inner, bitcoin_chain_tip)
+                let id = StateMachineId::from(&chain_tip.block_hash);
+                self.relay_message(id, msg.txid, &msg.inner, &chain_tip.block_hash)
                     .await?;
             }
             WstsNetMessage::DkgPublicShares(dkg_public_shares) => {
                 tracing::info!(
                     signer_id = %dkg_public_shares.signer_id,
                     "handling DkgPublicShares",
                 );
-                let id = StateMachineId::from(bitcoin_chain_tip);
+                let id = StateMachineId::from(&chain_tip.block_hash);
                 self.validate_sender(&id, dkg_public_shares.signer_id, &msg_public_key)?;
-                self.relay_message(id, msg.txid, &msg.inner, bitcoin_chain_tip)
+                self.relay_message(id, msg.txid, &msg.inner, &chain_tip.block_hash)
                     .await?;
             }
             WstsNetMessage::DkgPrivateShares(dkg_private_shares) => {
                 tracing::info!(
                     signer_id = %dkg_private_shares.signer_id,
                     "handling DkgPrivateShares"
                 );
-                let id = StateMachineId::from(bitcoin_chain_tip);
+                let id = StateMachineId::from(&chain_tip.block_hash);
                 self.validate_sender(&id, dkg_private_shares.signer_id, &msg_public_key)?;
-                self.relay_message(id, msg.txid, &msg.inner, bitcoin_chain_tip)
+                self.relay_message(id, msg.txid, &msg.inner, &chain_tip.block_hash)
                     .await?;
             }
```

Comment:
IIUC here (and in `DkgPublicShares`) we are now loading the "current" (chain tip wise) state machine, even for messages coming from a stale signer (eg, sending a message for an old chain tip).
I'm wondering if we should just ignore messages that are not from our chain tip directly before calling `handle_wsts_message`. Otherwise, we could also load the correct state machine, but at that point I'm not sure we care about it, and a rouge signer could mess with the LRU ordering (not an issue now, but it could be if we start to do signature rounds concurrently I guess).

Reply:
Yeah, good point. Fixed up in https://github.com/stacks-network/sbtc/pull/1221/commits/7be5559387c22ae874c582fdfd5ff04036584275.